pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -359,6 +359,27 @@ func GrowCluster(l *logger.Logger, c *Cluster, NumNodes int) error {
 	})
 }
 
+// ShrinkCluster removes tail nodes from an existing cluster.
+func ShrinkCluster(l *logger.Logger, c *Cluster, numNodes int) error {
+	providers := c.Clouds()
+	if len(providers) != 1 && providers[0] != gce.ProviderName {
+		return errors.Errorf("cluster %s is not on gce, shrinking a cluster is currently only supported on %s",
+			c.Name, gce.ProviderName)
+	}
+
+	if numNodes >= len(c.VMs) {
+		return errors.Errorf("cannot shrink cluster %s to %d nodes, only %d nodes in cluster",
+			c.Name, len(c.VMs)-numNodes, len(c.VMs))
```

Comment:
Are we okay with the log 
"cannot shrink cluster c to -3 nodes, only 5 nodes in cluster" when `len(c.VMs)` is 5 and `numNodes` is 8

Maybe a suggestion to `grow` the cluster is more appropriate?

Reply:
Sure, it might be odd. I considered changing it to `cannot shrink cluster by %d nodes ...` before - which may make more sense if a number larger than the node count is passed. Went with that idea, updated.